packages/icon/src/create-icon.tsx

```diff
@@ -32,10 +32,14 @@ export function createIcon(options: CreateIconOptions) {
   const {
     viewBox = "0 0 24 24",
     d: pathDefinition,
-    path,
     displayName,
     defaultProps = {},
   } = options
+  let path = options.path
+
+  if (Array.isArray(path)) {
+    path = path.map((p, index) => React.cloneElement(p, { key: index }))
+  }
```

Comment:
I think we can use React's internals to resolve this. Let's use `React.Children.toArray` instead.

```suggestion
  const path = React.Children.toArray(options.path)
```

Reply:
while it is an array now, the mapping to add the key appears to be missing :)

Reply:
interesting, TIL [code](https://github.com/facebook/react/blob/cae635054e17a6f107a39d328649137b83f25972/packages/react/src/ReactChildren.js#L112) - looks like this is good to go then!